lib/ansible/modules/packaging/os/zypper_repository.py

```diff
@@ -301,6 +318,7 @@ def main():
             name=dict(required=False),
             repo=dict(required=False),
             state=dict(choices=['present', 'absent'], default='present'),
+            extra_args_precommand=dict(required=False, default=None),
```

Comment:
Do not add `required=False` or `default=None` as these are implied. But please do add the parameter type explicitly.
```suggestion
            extra_args_precommand=dict(type='str'),
```
PS Yes, the other definitions are incorrect too, they should all include type information.

Reply:
@dagwieers Could there be a semi-automated cleanup of unnecessary `required` and `default` statements over all modules? Slow-changing ones like `zypper_repository` could benefit from that.

Reply:
@dagwieers I used the same pattern as zypper module does. But I will change it